### PR TITLE
Arizona 2023 Resident Personal Income Tax Parameter Updates

### LINKED ISSUE #3677: Arizona 2023 Resident Personal Income Tax Parameter Updates
According to the Form 140A, Arizona Resident Personal Income Tax Booklet, 2023 Version, there're some parameter updates.

Reference: https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet

### PR DESCRIPTION
**Pull request by Kevintowne**
Fixes #3677

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Arizona Resident Personal Income Tax Parameter Update.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Arizona 2023 income tax parameters.
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml:1]: do not delete any of the 2021 and 2022 references or values - just add the 2023 values and references

File: policyengine_us/parameters/gov/states/az/tax/income/main/joint.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/main/joint.yaml:1]: add references to all parameters that you update

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: add 2023 tests

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml [hunk 2]
```diff
@@ -10,18 +10,25 @@ metadata:
...
   2021-01-01: 25_100
   2022-01-01: 25_900
+  2023-01-01: 27_700
 HEAD_OF_HOUSEHOLD:
   2021-01-01: 18_800
   2022-01-01: 19_400
+  2023-01-01: 20_800
 WIDOW:
   2021-01-01: 25_100
   2022-01-01: 25_900
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml:26]: Did they change the widow filing rules? In this tax form they should be head of household, not joint (page 9).

image

We should make an `az_filing_status` variable that incorporates this logic.

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml [hunk 3]
```diff
@@ -10,18 +10,25 @@ metadata:
     href: https://azdor.gov/forms/individual/form-140-arizona-resident-personal-income-tax-booklet
   - title: 2021 Arizona Form 140 Resident Personal Income Tax Booklet
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf#page=43
+  - title: 2023 Arizona Form 140 Resident Personal Income Tax Booklet
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml:13]:
```suggestion
  - title: 2023 Arizona Form 140 Resident Personal Income Tax Booklet, Page 3
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/increased/rate.yaml [hunk 4]
```diff
@@ -10,7 +10,10 @@ metadata:
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf#page=43
   - title: 43-1041. Optional standard deduction Line I
     href: https://www.azleg.gov/viewdocument/?docName=https://www.azleg.gov/ars/43/01041.htm
+  - title: 2023 Arizona Form 140 Resident Personal Income Tax Booklet
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/increased/rate.yaml:13]:
```suggestion
  - title: 2023 Arizona Form 140 Resident Personal Income Tax Booklet, Page 3
```

File: policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml [hunk 5]
```diff
@@ -11,29 +11,38 @@ metadata:
       href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140BOOKLET.pdf
     - title: 2021 Arizona Form 140 Resident Personal Income Tax Booklet
       href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf 
+    - title: 2023 Arizona Form 140 Resident Personal Income Tax Booklet
+      href: https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet
+
 brackets:
   - threshold:
       2019-01-01: 0
+      2023-01-01: 0
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml:20]: can remove

File: policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml [hunk 6]
```diff
@@ -11,29 +11,38 @@ metadata:
       href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140BOOKLET.pdf
     - title: 2021 Arizona Form 140 Resident Personal Income Tax Booklet
       href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf 
+    - title: 2023 Arizona Form 140 Resident Personal Income Tax Booklet
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml:14]:
```suggestion
    - title: 2023 Arizona Form 140 Resident Personal Income Tax Booklet, Page 3
```

Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml:14]: Can you add a legislative reference? Looks like the legal code isn't updated to reflect the 2.5% flat tax so we may need to locate a bill.

File: policyengine_us/tests/policy/baseline/gov/states/az/tax/income/az_income_tax_before_non_refundable_credits.yaml [hunk 7]
```diff
@@ -14,7 +14,7 @@
     filing_status: JOINT
     az_taxable_income: 60_000
   output:
-    az_income_tax_before_non_refundable_credits: 1541.5885
+    az_income_tax_before_non_refundable_credits: 1500
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/az/tax/income/az_income_tax_before_non_refundable_credits.yaml:17]: And elsewhere
```suggestion
    az_income_tax_before_non_refundable_credits: 1_500
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml [hunk 8]
```diff
@@ -22,6 +23,3 @@ WIDOW:
 SINGLE:
   2021-01-01: 12_550
   2022-01-01: 12_950
-SEPARATE:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml:25]: deleted separate but need to delete widow

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml [hunk 9]
```diff
@@ -4,12 +4,13 @@ metadata:
   period: year
   unit: currency-USD
   breakdown:
-  - filing_status
+  - az_filing_status
   reference:
   - title: 2022 Arizona Form 140 Resident Personal Income Tax Booklet
     href: https://azdor.gov/forms/individual/form-140-arizona-resident-personal-income-tax-booklet
   - title: 2021 Arizona Form 140 Resident Personal Income Tax Booklet
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf#page=43
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml:12]: add 2023 values + reference

File: policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml [hunk 10]
```diff
@@ -34,6 +37,3 @@ brackets:
       2019-01-01: 318_000
       2021-01-01: 333_684
       2022-01-01: .inf
-    # in 2021 Arizona broke the highest income tax rate up into a 1.00% base rate and 3.5% surcharge
-    rate:
-      2019-01-01: 0.045
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml:39]: I dont think this is correct

File: policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml [hunk 11]
```diff
@@ -11,6 +11,9 @@ metadata:
       href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140BOOKLET.pdf
     - title: 2021 Arizona Form 140 Resident Personal Income Tax Booklet
       href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf 
+    - title: ARIZONA STATE SENATE, Fifty-Fifth Legislature, First Regular Session, Paragraph 1,Purpose 
+      href: https://www.azleg.gov/legtext/55leg/1R/summary/S.1828APPROP.DOCX.htm
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml:15]: add 2023 tax form

File: policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml [hunk 12]
```diff
@@ -11,6 +11,9 @@ metadata:
...
+
 brackets:
   - threshold:
       2019-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml:19]: add 2023 values

File: policyengine_us/parameters/gov/states/az/tax/income/main/joint.yaml [hunk 13]
```diff
@@ -34,6 +36,4 @@ brackets:
       2019-01-01: 318_000
       2021-01-01: 333_684
       2022-01-01: .inf
-    # in 2021 Arizona broke the highest income tax rate up into a 1.00% base rate and 3.5% surcharge
-    rate:
-      2019-01-01: 0.045
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/main/joint.yaml:39]: check all the files -> should not have deleted the last lines of code

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: add the 2023 tax form to ALL of the az parameters even if the value is the same

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml [hunk 14]
```diff
@@ -4,12 +4,13 @@ metadata:
   period: year
   unit: currency-USD
   breakdown:
-  - filing_status
+  - az_filing_status
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml:7]: check other AZ files with filing status and delete widow

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml [hunk 15]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Arizona 2023 income tax parameters.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Arizona 2023 income tax parameters. 
    - Arizona filing status variable.
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: property tax form: 
https://azdor.gov/forms/tax-credits-forms/property-tax-refund-credit-claim-form-fillable

File: policyengine_us/parameters/gov/states/az/tax/income/credits/family_tax_credits/amount/cap.yaml [hunk 16]
```diff
@@ -22,3 +22,5 @@ reference:
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140i-2D.pdf#page=23
   - title: 2021 Arizona Tax Form Instructions Line50
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140i-2D.pdf#page=21
+  - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet Line22
+    href: https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet#page=21
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/credits/family_tax_credits/amount/cap.yaml:26]: should be page 22

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/tax/income/credits/family_tax_credits/amount/cap.yaml:26]: check all yaml files

File: policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py [hunk 17]
```diff
@@ -0,0 +1,41 @@
+from policyengine_us.model_api import *
+
+
+class StateFilingStatusIfMarriedFilingSeparatelyOnSameReturn(Enum):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py:4]:
```suggestion
class ArizonaFilingStatus(Enum):
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: delete widow files

File: policyengine_us/parameters/gov/states/az/tax/income/credits/dependent_credit/reduction/start.yaml [hunk 18]
```diff
@@ -18,6 +18,8 @@ metadata:
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140i-2D.pdf#page=20
   - title: Arizona Form 140 - 2022 Resident Personal Income Tax Return
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140i-2D.pdf#page=21
+  - title: Arizona Form 140A - 2023 Resident Personal Income Tax Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/credits/dependent_credit/reduction/start.yaml:21]: add page in the title

File: policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+    SINGLE = "Single"
+    SEPARATE = "Separate"
+    HEAD_OF_HOUSEHOLD = "Head of household"
+    WIDOW = "Widow(er)"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py:8]:
```suggestion
```

File: policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+    JOINT = "Joint"
+
+
+class az_filing_status(Variable):
+    value_type = Enum
+    entity = TaxUnit
+    possible_values = StateFilingStatusIfMarriedFilingSeparatelyOnSameReturn
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py:15]: adjust accordingly

File: policyengine_us/parameters/gov/states/az/tax/income/exemptions/parent_grandparent/cost_rate.yaml [hunk 21]
```diff
@@ -4,6 +4,8 @@ metadata:
   period: year
   label: Arizona parents and grandparents exemption cost rate 
   reference: 
+    - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet line11a
+      href: https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet#page=10
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/exemptions/parent_grandparent/cost_rate.yaml:8]:
```suggestion
      href: https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/tax/income/exemptions/parent_grandparent/cost_rate.yaml:8]: does not work for non .pdf links - need to add page to title

File: policyengine_us/parameters/gov/states/az/tax/income/exemptions/parent_grandparent/cost_rate.yaml [hunk 22]
```diff
@@ -4,6 +4,8 @@ metadata:
   period: year
   label: Arizona parents and grandparents exemption cost rate 
   reference: 
+    - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet line11a
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/exemptions/parent_grandparent/cost_rate.yaml:7]:
```suggestion
    - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet Page 10, Line 11a
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/tax/income/exemptions/parent_grandparent/cost_rate.yaml:7]: same style in all yaml without hyperlink

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+        return select(
+            [
+                us_filing_status == fsvals.JOINT,
+                us_filing_status == fsvals.SINGLE,
+                us_filing_status == fsvals.SEPARATE,
+                us_filing_status == fsvals.HEAD_OF_HOUSEHOLD,
+                us_filing_status == fsvals.WIDOW,
+            ],
+            [
+                # Simulate scenario where joint filers file separately on Arkansas return.
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py:31]:
```suggestion
                # In Arkansas, widowed filers are treated as heads of households.
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml [hunk 24]
```diff
@@ -4,24 +4,30 @@ metadata:
...
   2022-01-01: 25_900
 SINGLE:
   2021-01-01: 12_550
   2022-01-01: 12_950
-SEPARATE:
-  2021-01-01: 12_550
-  2022-01-01: 12_950
+  2023-01-01: 13_850
+SEPERATE:
+  2023-01-01: 13_850
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml:33]: add back the 2021 and 2022 values

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/increased/rate.yaml (hunk 4 above)
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/increased/rate.yaml:13]: add pages to the title if there is no hyperlink

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/increased/rate.yaml:13]: all yaml files

File: policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml [hunk 25]
```diff
@@ -7,29 +7,37 @@ metadata:
   reference:
     - title: Arizona State Legislature Title 43 - Taxation of Income, Chapter 10, Article 5, 43-1011
       href: https://www.azleg.gov/viewdocument/?docName=https://www.azleg.gov/ars/43/01011.htm
+    - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet
+      href: https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet
     - title: 2022 Arizona Form 140 Resident Personal Income Tax Booklet
       href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140BOOKLET.pdf
     - title: 2021 Arizona Form 140 Resident Personal Income Tax Booklet
       href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf 
+    - title: ARIZONA STATE SENATE, Fifty-Fifth Legislature, First Regular Session, Paragraph 1,Purpose
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml:16]: add section to law source

File: policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml [hunk 26]
```diff
@@ -7,29 +7,37 @@ metadata:
...
   - threshold:
       2019-01-01: 106_000
       2021-01-01: 111_229
       2022-01-01: .inf
     rate:
       2019-01-01: 0.0417
+      2023-01-01: 0.025
   - threshold:
       2019-01-01: 318_000
       2021-01-01: 333_684
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/main/head_of_household.yaml:43]: add 2.5% to the bottom rate - all main files

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml [hunk 27]
```diff
@@ -4,24 +4,32 @@ metadata:
...
+
 JOINT:
   2021-01-01: 25_100
   2022-01-01: 25_900
+  2023-01-01: 27_700
 HEAD_OF_HOUSEHOLD:
   2021-01-01: 18_800
   2022-01-01: 19_400
+  2023-01-01: 20_800
 WIDOW:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/standard/amount.yaml:25]: delete widow here

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/capital_gains/rate.yaml [hunk 28]
```diff
@@ -6,6 +6,8 @@ metadata:
   reference:
   - title: 43-1022. Subtractions from Arizona gross income (22)
     href: https://www.azleg.gov/ars/43/01022.htm
+  - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/capital_gains/rate.yaml:9]: add page

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/college_savings/cap.yaml [hunk 29]
```diff
@@ -4,20 +4,20 @@ metadata:
...
 JOINT:
   2021-01-01: 4_000
-WIDOW:
-  2021-01-01: 2_000
 SEPARATE:
   2021-01-01: 4_000
 reference:
   - title: 43-1022. Subtractions from Arizona gross income (27)
     href: https://www.azleg.gov/ars/43/01022.htm
+  - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/college_savings/cap.yaml:19]: add page

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/military_retirement/max_amount.yaml [hunk 30]
```diff
@@ -11,6 +11,8 @@ metadata:
   reference:
     - title: Subtractions from Arizona gross income 43-1022(26) 
       href: https://www.azleg.gov/ars/43/01022.htm
+    - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/military_retirement/max_amount.yaml:14]: add page

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/pension/public_pension_cap.yaml [hunk 31]
```diff
@@ -8,6 +8,8 @@ metadata:
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140BOOKLET.pdf#page=18
   - title: 2021 Arizona Form 140 Resident Personal Income Tax Booklet
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf#page=23
+  - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/pension/public_pension_cap.yaml:11]: add page

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/subtractions.yaml [hunk 32]
```diff
@@ -13,6 +13,8 @@ metadata:
   period: year
   label: Arizona adjusted gross income subtractions
   reference:
+    - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/subtractions.yaml:16]: add page

File: policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py [hunk 33]
```diff
@@ -0,0 +1,38 @@
...
+    entity = TaxUnit
+    possible_values = ArizonaFilingStatus
+    default_value = ArizonaFilingStatus.SINGLE
+    definition_period = YEAR
+    label = "Arizona filing status"
+
+    def formula(tax_unit, period, parameters):
+        us_filing_status = tax_unit("filing_status", period)
+        fsvals = us_filing_status.possible_values
+        return select(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py:22]: this needs another test (probably for widow input)

File: policyengine_us/parameters/gov/states/az/tax/income/credits/family_tax_credits/amount/cap.yaml [hunk 34]
```diff
@@ -22,3 +22,5 @@ reference:
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140i-2D.pdf#page=23
   - title: 2021 Arizona Tax Form Instructions Line50
     href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140i-2D.pdf#page=21
+  - title: 2023 Arizona Form 140A - Arizona Resident Personal Income Tax Booklet Line22 Step 4
+    href: https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet#page=22
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/credits/family_tax_credits/amount/cap.yaml:26]:
```suggestion
    href: https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/tax/income/credits/family_tax_credits/amount/cap.yaml:26]: page in title the #page= only works for .pdf files

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Comment by PavelMakarchuk** [comment]
Could you add a comment to the issue describing all parameters that changed

File: policyengine_us/tests/policy/baseline/gov/states/az/tax/income/az_filing_status.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/az/tax/income/az_filing_status.yaml:1]: add one more test

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py [hunk 35]
```diff
@@ -0,0 +1,39 @@
...
+    JOINT = "Joint"
+
+
+class az_filing_status(Variable):
+    value_type = Enum
+    entity = TaxUnit
+    possible_values = ArizonaFilingStatus
+    default_value = ArizonaFilingStatus.SINGLE
+    definition_period = YEAR
+    label = "Arizona filing status"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py:17]: Add a legal code reference indicating that surviving spouses are treated as HoH

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py [hunk 36]
```diff
@@ -0,0 +1,38 @@
...
+    JOINT = "Joint"
+
+
+class az_filing_status(Variable):
+    value_type = Enum
+    entity = TaxUnit
+    possible_values = ArizonaFilingStatus
+    default_value = ArizonaFilingStatus.SINGLE
+    definition_period = YEAR
+    documentation = "https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet"  # Page 5, Box 5
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py:17]:
```suggestion
    documentation = (
        # Section 43-1001(5)
        "https://www.azleg.gov/viewdocument/?docName=https://www.azleg.gov/ars/43/01001.htm",
        # Page 5 (page 9 of pdf), Box 5.
        "https://azdor.gov/forms/individual/form-140a-arizona-resident-personal-income-tax-booklet"
    )
```

**Review by MaxGhenis — APPROVED** [review]

File: policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py (hunk 35 above)
Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/az/tax/income/az_filing_status.py:18]: Please note I requested legal code not tax form

**Pull request merged**